src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -93,7 +94,7 @@ public virtual IEnumerable<ICompletedFakeObjectCall> RecordedCallsInScope
             get { return FakeScope.Current.GetCallsWithinScope(this); }
         }
 
-        internal List<ICompletedFakeObjectCall> AllRecordedCalls
+        internal IEnumerable<ICompletedFakeObjectCall> AllRecordedCalls
         {
             get { return this.recordedCallsField; }

```

Comment:
@thomaslevesque, I think the "real" `ConcurrentQueue<T>` has better performance characteristics. It's supposed to be [super fancy and lockless and all kinds of stuff](https://msdn.microsoft.com/en-us/library/dd997305%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396). For our purposes, it probably doesn't matter. We could use all "shim". Me, I like the idea of, if ever we drop 3.5, just deleting the project and then we have less of our own code.

I think the `ToList` and the property/method concerns were separate in @adamralph's mind. I think the property/method concern comes down to this:

> If it is a method that is mutating the internal state, I prefer a method to retrieve that mutated state.

Currently I have a method mutating internal state that is then accessed via a property. I see the point, but feel a little less strongly about it than @adamralph, it seems. However, like you, I don't mind it being a method (in fact, @adamralph has swayed me, and I think it's a little better, but I'd go with the prevailing flow).

However, I am with you on the `ToList` front. I think it's excessive for a property (soon method) that exposes the field as an `IEnumerable` and that is `internal`. So I think omit it. @adamralph, please continue trying to convince us otherwise if you think it's warranted.